src/provider/mod.rs

```diff
@@ -885,23 +884,17 @@ async fn send_blob<W: AsyncWrite + Unpin + Send + 'static>(
             size,
         })) => {
             write_response(&mut writer, buffer, Res::Found).await?;
-            // need to thread the writer though the spawn_blocking, since
-            // taking a reference does not work. spawn_blocking requires
-            // 'static lifetime.
-            writer = tokio::task::spawn_blocking(move || {
-                let file_reader = std::fs::File::open(&path)?;
-                let outboard_reader = std::io::Cursor::new(outboard);
-                let mut wrapper = SyncIoBridge::new(&mut writer);
-                let mut slice_extractor = abao::encode::SliceExtractor::new_outboard(
-                    file_reader,
-                    outboard_reader,
-                    0,
-                    size,
-                );
-                let _copied = std::io::copy(&mut slice_extractor, &mut wrapper)?;
-                std::io::Result::Ok(writer)
-            })
-            .await??;
+
+            let outboard =
+                PreOrderMemOutboard::new(name.into(), IROH_BLOCK_SIZE, outboard.to_vec());
```

Comment:
is there a way to avoid this `to_vec` call?

Reply:
Ah, yes. I even did this on one of the branches that build on this. Just need to cherry pick it.

There are outboard versions that work with a &[u8] (and can't write), just need to use those.